### PR TITLE
TWCoinTypeConfigTests reorganization

### PR DESCRIPTION
**Pull request by catenocrypt**
Issue #694 .
1st commit:  Tests refreshed/completed to be in sync with current coins.json.  Generated using a helper script.  Verified in PR.
2nd commit:  Tests reorganized, one test per coin (56 of them), each containing all the coin property calls (about 8).  ('56 tests from TWCoinTypeConfiguration')
3rd commit: The test file TWCoinTypeConfigTests.cpp has been split up into 56 files, one per coin, tests/interface/TW*CoinTypeTests.cpp 

Note: this PR would break in-progress new coin PRs.

### CODE REVIEW

**Comment by hewigovens** [comment]
we can also add the helper script to `codegen` folder

**Comment by catenocrypt** [comment]
The test file TWCoinTypeConfigTests.cpp has been split up into 56 files, one per coin, tests/interface/TW*CoinTypeTests.cpp

**Comment by catenocrypt** [comment]
The generator script is a Python3 script.  It is NOT intended to be used at every build, because that would make the tests totally meaningless (both code and its unit tests generated every time from the same input).
I can add the script to 'tools' folder (rather than codegen), but I'm still hesitating, it would probably get obsolete soon and just lay there unused.

**Comment by vikmeup** [comment]
@catenocrypt any chance we can use ruby for consistency? as it's used for codegen

**Comment by hewigovens** [comment]
The tests are only meaningful in the app, we can put the script in the `tools` folder and it would be easy to update some items next time

File: tests/interface/TWARKCoinTypeTests.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "TWTestUtilities.h"
+
```
Comment (hewigovens) [inline tests/interface/TWARKCoinTypeTests.cpp]: just coding styles, let's remove empty lines between include

File: tests/interface/TWARKCoinTypeTests.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include <TrustWalletCore/TWCoinTypeConfiguration.h>
+
+#include <gtest/gtest.h>
+
+
+// Originally one-time generated by script gen_test_cointype.py
+
+TEST(TWCoinTypeConfiguration, TWCoinTypeARK) {
+    auto symbol = WRAPS(TWCoinTypeConfigurationGetSymbol(TWCoinTypeARK));
```
Comment (hewigovens) [inline tests/interface/TWARKCoinTypeTests.cpp]: can we group by declarations / assert? easy to read

```c++
auto symbol = ...
auto txId = ...

assertStringsEqual(...)
assertStringsEqual(...)

ASSERT_EQ(...)
```

**Comment by hewigovens** [comment]
can we move them to specific blockchain folder? like `tests/Aeternity/TWCoinTypeTests`

File: codegen/lib/templates/TWCoinTypeTests.cpp.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+TEST(TW<%= format_name(coin['name']) %>CoinType, TWCoinType) {
+    auto symbol = WRAPS(TWCoinTypeConfigurationGetSymbol(TWCoinType<%= format_name(coin['name']) %>));
+    auto txId = TWStringCreateWithUTF8Bytes("123");
+    auto txUrl = WRAPS(TWCoinTypeConfigurationGetTransactionURL(TWCoinType<%= format_name(coin['name']) %>, txId));
+    auto id = WRAPS(TWCoinTypeConfigurationGetID(TWCoinType<%= format_name(coin['name']) %>));
+    auto name = WRAPS(TWCoinTypeConfigurationGetName(TWCoinType<%= format_name(coin['name']) %>));
+
+    assertStringsEqual(symbol, "<%= coin['symbol'] %>");
+    ASSERT_EQ(TWCoinTypeConfigurationGetDecimals(TWCoinType<%= format_name(coin['name']) %>), <%= coin['decimals'] %>);
```
Comment (hewigovens) [inline codegen/lib/templates/TWCoinTypeTests.cpp.erb]: can we put this `ASSERT_EQ` along with other `ASSERT_EQ`s?

Comment (catenocrypt) [reply codegen/lib/templates/TWCoinTypeTests.cpp.erb]: Done :)

**Pull request merged**